### PR TITLE
Update smoke tests to use requests instead of seleium.

### PR DESCRIPTION
**Pull request by jrbenny35**
More fixes for the smoke test suite. This adds a selenium-firefox docker image that the tests can run in.

### CODE REVIEW

File: tests/smoke/setup_docker.sh [hunk 1]
```diff
@@ -11,4 +11,6 @@ docker-compose -f addons-server/docker-compose.yml -f tests/ui/docker-compose.fu
 docker-compose -f addons-server/docker-compose.yml -f tests/ui/docker-compose.functional-tests.yml restart web
 docker-compose -f addons-server/docker-compose.yml -f tests/ui/docker-compose.functional-tests.yml exec web make setup-ui-tests
 docker-compose -f addons-server/docker-compose.yml -f tests/ui/docker-compose.functional-tests.yml stop nginx
-docker-compose -f addons-server/docker-compose.yml -f tests/ui/docker-compose.functional-tests.yml up -d
\ No newline at end of file
+docker-compose -f addons-server/docker-compose.yml -f tests/ui/docker-compose.functional-tests.yml up -d
+# Pull and run selenium firefox docker image
+docker run -p 4444:4444 --detach selenium/standalone-firefox
```
Comment (kumar303) [inline tests/smoke/setup_docker.sh]: This works for me, as in, I can curl `localhost:4444` and see some output. 

However, I am a bit stuck in trying to test out the tests locally. I nuked all of my containers and images so I could start over. Now, I get: ERROR: pull access denied for addons-frontend, repository does not exist or may require 'docker login'. It does not build the image automatically.

I'm not really sure what's going on but I guess you might as well merge this so you can test on circle CI.

**Comment by kumar303** [comment]
I think it needs this change to build images automatically:

```diff
diff --git a/tests/ui/docker-compose.functional-tests.yml b/tests/ui/docker-compose.functional-tests.yml
index 288096ae8..ee1db34bd 100644
--- a/tests/ui/docker-compose.functional-tests.yml
+++ b/tests/ui/docker-compose.functional-tests.yml
@@ -5,5 +5,6 @@ version: "2.3"
 
 services:
   addons-frontend:
-    image: addons-frontend:latest
     # This should build a local image if it isn't built already.
+    build: ../.
+    image: addons-frontend:latest
```

**Comment by jrbenny35** [comment]
> Actually, I changed my mind on this one. I just noticed a problem where the selenium docker image was caching the network connection. Selenium seems like a layer of complexity that we don't need to deal with here. I'd rather see a more direct approach. Here is my suggestion, which is working locally for me.

I think maybe we can just use selenium on circleci, and leave the local setup to the user.  But looking at your solution, we may not even need a browser. We could use something like beautifulsoup to scrape the page. I'd prefer that over using urllib directly.

**Comment by kumar303** [comment]
> I'd prefer that over using urllib directly.

Why? It just needs to check if the comment appears in the output. A py.test failure with urllib isn't super great but I don't think beautiful soup would make it any better.

```
>       assert "<!-- Godzilla NOPE of browsers -->" in page_source
E       assert '<!-- Godzilla NOPE of browsers -->' in '<!DOCTYPE html>\n<html lang="en-US" dir="ltr" data-reactroot=""><head><title data-react-helmet="true">Add-ons for Fir...pe="application/json">[]</script><script src="//olympia.test:3001/amo-8eae868180d3dd92175a.js"></script></body></html>'
```

**Comment by jrbenny35** [comment]
> Why? It just needs to check if the comment appears in the output. A py.test failure with urllib isn't super great but I don't think beautiful soup would make it any better.

Looking at your code, there isn't any error protection. What if it times out? I am sure beautifulsoup, which uses urllib, has some error handling that we don't have to figure out ourself

**Comment by willdurand** [comment]
> Looking at your code, there isn't any error protection. What if it times out? I am sure beautifulsoup, which uses urllib, has some error handling that we don't have to figure out ourself

FWIW, I like the simple approach with only `urllib`. We don't need complicated. I think `curl | grep` would work too, it's a smoke test, it does not have to be complicated :)

**Comment by jrbenny35** [comment]
So in the effort of maintenance, as that would probably fall onto me, I think requests is a good compromise. It is only a few lines of code, compared to urllib. Runs just as fast too.

**Comment by kumar303** [comment]
Please update the title of the PR and / or the final commit message before merging.

**Pull request merged**